### PR TITLE
Feature/tcin 60

### PR DESCRIPTION
**Pull request by GTerigiHastega**
Caching of POST, PUT and PATCH requests implemented. Need to refactor common code and further expand each and every method.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -21,17 +19,21 @@ import https from 'https';
 import WebSocket from 'ws';
 import config from 'config';
 
-if (process.env.NODE_ENV === 'development') {
-    const httpsAgent = new https.Agent({
-        rejectUnauthorized: false,
-    });
+// Manager
```
Comment (pablomichelettii) [inline src/app.ts:22]: Don't use repetitve comments. If u use self explanatory naming convention, as u have done, is enough

File: src/app.ts [hunk 2]
```diff
@@ -21,17 +19,21 @@ import https from 'https';
...
+import processEnvManager from './manager/processEnv/processEnv.manager';
+
+// To discuss
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const swaggerDocument = require('./swagger-output.json');
```
Comment (pablomichelettii) [inline src/app.ts:28]: I suppose the problem is that u're trying to use a json importing it as a module. If not let's discuss what is the problem @Zenomante

File: src/app.ts [hunk 3]
```diff
@@ -21,17 +19,21 @@ import https from 'https';
...
 
-    axios.defaults.httpsAgent = httpsAgent;
-    console.log(process.env.NODE_ENV, `RejectUnauthorized is disabled.`);
+if (processEnvManager.isDevelopment()) {
+    axios.defaults.httpsAgent = new https.Agent({ rejectUnauthorized: false });
+    console.log('development', `RejectUnauthorized is disabled.`);
```
Comment (pablomichelettii) [inline src/app.ts]: could we make this log dynamic?

File: src/controllers/localCache.controller.ts [hunk 4]
```diff
@@ -1,131 +1,121 @@
...
-        }
+    const successResponseBody = new BodyResponseModel();
 
-        return res.status(200).send({
-            error: false,
-            cache: cacheData,
-            message: messageData,
-            data: responseData,
+    if (!noCache && existsSync(directoryPath) && existsSync(currentFile)) {
+        console.log(currentFile);
```
Comment (pablomichelettii) [inline src/controllers/localCache.controller.ts]: remove log

File: src/controllers/localCache.controller.ts [hunk 5]
```diff
@@ -1,131 +1,121 @@
...
+        );
+        promises.mkdir(directoryPath, { recursive: true }).then((_) => {
+            if (!noCache) writeFileSync(currentFile, JSON.stringify(fetch.data));
         });
+        successResponseBody.data = fetch.data;
+        successResponseBody.message = "here's the fetched data";
+        successResponseBody.cache = false;
+        successResponseBody.filenameCached = currentFile;
     }
+    console.log(successResponseBody);
```
Comment (pablomichelettii) [inline src/controllers/localCache.controller.ts]: remove console log

**Review by pablomichelettii — CHANGES_REQUESTED** [review]
that's nice! GG. For the next tasks remember to update the version of Loki (@Zenomante @GTerigiHastega )

**Comment by GTerigiHastega** [comment]
Logging utility added. Increased version to 1.0.01

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "loki",
-  "version": "1.0.0",
+  "version": "1.0.01",
```
Comment (pablomichelettii) [inline package.json]: don`t use 0 before the number, 1.0.1 it's ok

File: package.json [hunk 7]
```diff
@@ -35,10 +37,13 @@
     "morgan": "^1.10.0",
     "nanoid": "^3.3.1",
     "node-typescript": "^0.1.3",
+    "npmlog": "^7.0.1",
```
Comment (pablomichelettii) [inline package.json]: you can remove it if u use wiston

File: src/app.ts [hunk 8]
```diff
@@ -21,17 +19,22 @@ import https from 'https';
...
 
-if (process.env.NODE_ENV === 'development') {
-    const httpsAgent = new https.Agent({
-        rejectUnauthorized: false,
-    });
+// Manager
+import processEnvManager from './manager/processEnv/processEnv.manager';
+import { logger } from './utils/logger';
+
+// To discuss
```
Comment (pablomichelettii) [inline src/app.ts]: Choose a common tag for those comments: for example i'm pretty sure there are some vscode extensions that recognize tags like //TODO: or //@TODO: just chose one so we can start to use one of those tags

File: src/server.ts [hunk 9]
```diff
@@ -1,8 +1,11 @@
 import app from './app';
 import { createConnection } from './db';
+import { logger, clearLogs } from './utils/logger';
 
 createConnection();
 
 app.listen(app.get('port'), () => {
-    console.log('Server on port', app.get('port'));
+    // console.log('Server on port', app.get('port'));
```
Comment (pablomichelettii) [inline src/server.ts:8]: duplicate

**Review by pablomichelettii — COMMENTED** [review]
GGWD

**Review by pablomichelettii — COMMENTED** [review]
LGTM

**Pull request merged**